src/bernstein.jl

```diff
@@ -0,0 +1,108 @@
+export BernsteinBasis
+
+struct BernsteinBasis{N} <: AbstractP4MLBasis where {N} end
+
+BernsteinBasis(N::Integer) = BernsteinBasis{N}()
+
+Base.length(basis::BernsteinBasis{N}) where {N} = N
+
+natural_indices(basis::BernsteinBasis) = [ (n = n,) for n = 0:length(basis)-1 ]
+
+_valtype(basis::BernsteinBasis, T::Type{<:Real}) = T
+
+_generate_input(basis::BernsteinBasis) = 2 * rand() - 1 
+
+@generated function static_binomial(::Val{N}) where {N}
```

Comment:
looks ok. I now also think a simple `ntuple` might work, we can experiment with that later.